### ISSUE TITLE
Should context-manipulating functions return effectful unit instead of context?

### ISSUE THREAD

**Issue by aztecrex** [purescript-contrib/purescript-canvas#46]
The context-manipulating functions all return a context in `Eff`. However, the original
context is modified by the function. For do notation, this adds the noise of either
`void`-ing operations or capturing their values to underscores.  I understand there is
_purescript-free-canvas_ but that is not yet in _package-sets_ so I hesitate to use it.

In fact, because of the `Graphics.Canvas` API, I originally (wrongly) assumed that I
had to capture each context and pass it to the next operation. In this API, the types
don't tell the story which makes it harder to learn.

I propose that such operations return an `Eff Unit` instead of  `Eff Context2D`. That
is a breaking change, of course, so no PR yet.

**Comment by paf31** [comment]
Yes, I agree we should return `Unit` especially with the new `Discard` warnings in the compiler. We should break this in the next major release.

**Comment by jacereda** [comment]
Given a signature like:
```
setLineWidth :: forall eff. Number -> Context2D -> Eff (canvas :: CANVAS | eff) Context2D
```
I'd either:

- Retain current signature and add (note that `Context2D` is now the first argument):
```
setLineWidth_ :: forall eff. Context2D -> Number -> Eff (canvas :: CANVAS | eff) Unit
```

- Break compatibility and use the above as `setLineWidth`.

**Comment by jacereda** [comment]
I've been surveying usage of this module on github and didn't find instances where chained operation was used (via `>>=` I mean). So, I guess the second option wouldn't be too radical.

**Comment by hdgarrood** [comment]
Resolved by #53.
